content/develop/interact/search-and-query/advanced-concepts/vectors.md

```diff
@@ -9,369 +9,503 @@ categories:
 
-Optional parameters are:
+| Distance metric | Description | Mathematical representation |
+| --------------- | ----------- | --------------------------- |
+| `L2` | Euclidean distance between two vectors. | $d(u, v) = \sqrt{ \displaystyle\sum_{i=1}^n{(u_i - v_i)^2}}$ |
+| `IP` | Inner product of two vectors. | $d(u, v) = 1 -u\cdot v$ |
+| `COSINE` | Cosine distance of two vectors. | $d(u, v) = 1 -\frac{u \cdot v}{\lVert u \rVert \lVert v  \rVert}$ |
 
-* `INITIAL_CAP` - Initial vector capacity in the index affecting memory allocation size of the index.
+*The above metrics calculate distance between two vectors, where the smaller the value is, the closer the two vectors are in vector space.*
 
-* `M` - Number of maximum allowed outgoing edges for each node in the graph in each layer. on layer zero the maximal number of outgoing edges will be `2M`. Default is 16.
+## Store and update vectors
 
-* `EF_CONSTRUCTION` - Number of maximum allowed potential outgoing edges candidates for each node in the graph, during the graph building. Default is 200.
+On index creation, the `<storage_type>` dictates how vector and metadata are structured and loaded into Redis.
 
-* `EF_RUNTIME` - Number of maximum top candidates to hold during the KNN search. 
-Higher values of `EF_RUNTIME` lead to more accurate results at the expense of a longer runtime. Default is 10.
-
-* `EPSILON` - Relative factor that sets the boundaries in which a range query may search for candidates. That is, vector candidates whose distance from the query vector is `radius*(1 + EPSILON)` are potentially scanned, allowing more extensive search and more accurate results (on the expense of runtime). Default is 0.01.   
+### Hash
+Store or update vectors and any metadata in [hashes]({{< baseurl >}}/develop/data-types/hashes/) with the [`HSET`]({{< baseurl >}}/commands/hset/) command.
 
 **Example**
-
 ```
-FT.CREATE my_index2 
-SCHEMA vector_field VECTOR 
-HNSW 
-16 
-TYPE FLOAT64 
-DIM 128 
-DISTANCE_METRIC L2 
-INITIAL_CAP 1000000 
-M 40 
-EF_CONSTRUCTION 250 
-EF_RUNTIME 20
-EPSILON 0.8
+HSET docs:01 vector <vector_bytes> foo bar
 ```
 
-## Indexing vectors
+<br/>
 
-### Storing vectors in hashes
-Storing vectors in Redis hashes is done by sending a binary blob of vector data. A common way of doing so is by using python numpy with redis-py client:
+{{% alert title="Tip" color="warning" %}}
+Hashes expect all data to be represented as strings. Thus, `<vector_bytes>` represents a bytes version of the vector.
+{{% /alert  %}}
+
+<br/>
+
+A common way of converting vectors to bytes is through the Python [numpy](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tobytes.html) library and [redis-py](https://redis-py.readthedocs.io/en/stable/examples/search_vector_similarity_examples.html) client:
 ```py
 import numpy as np
 from redis import Redis
```

Comment:
something is a bit odd to me about jumping to a python example if all other example are just cli. I think the long term plan is to have each example be a carousel of client libary examples but if that's not till the future may consider making this just a note or link to something else

Reply:
Yeah, agreed. This was in place before, I just tried to improve the Python ;)